### ISSUE TITLE
C11 _Generic is not supported

### ISSUE THREAD

**Issue by chrysn** [immunant/c2rust#760]
When transpiled code contains C11's _Generic macros, this warns:

```
warning: c2rust: Encountered unsupported generic selection expression
```

Please consider adding support for this.

---

As I assume it's not easy to add, and RIOT-OS might be one of the few users (for many other use cases of C2Rust are about converting legacy code bases that don't use modern C features), I'll perfectly understand if this is a low priority issue -- in that case, the issue still serves tracking purposes (I can reference it in workarounds, and see that the workarounds can be removed when this is closed), and to collect any workaround pointers.

**Comment by GPHemsley** [comment]
https://github.com/immunant/c2rust/blob/516b30e3b632a29c4206dbdfe16130867e11e7a4/c2rust-ast-exporter/src/AstExporter.cpp#L1290-L1293
